plugins/woocommerce/client/legacy/js/admin/wc-enhanced-select.js

```diff
@@ -305,6 +305,81 @@ jQuery( function( $ ) {
 					$( this ).selectWoo( select2_args ).addClass( 'enhanced' );
 				});
 
+				function markMatch(text, term) {
+					var startString = '<span class="boldwrap">';
+					var endString = text.replace(startString, '');
+					var match = endString.toUpperCase().indexOf((term || '').toUpperCase());
+
+					if (match < 0) {
+						return text;
+					}
+					var elementToReplace = endString.substr(match, term.length);
+
+					var $match = '<strong>' + endString.substring(match, match + term.length) + '</strong>';
+
+					return startString + endString.replace(elementToReplace, $match);
+				}
+				// Ajax category search boxes
+				$( ':input.wc-category-select' ).filter( ':not(.enhanced)' ).each( function() {
+					var return_format = $( this ).data( 'return_id' ) ? 'id' : 'slug';
+					var query;
+
+					function formatResult(data) {
+						if (data.loading){
+							return data.text;
+						}
+						var markup = "";
+						var term = query.term || '';
+						var id = 'state' + data.term_id;
+						markup = $('<div class="checkbox"><input id="'+id+'" type="checkbox" '+(data.selected ? 'checked': '')+'><label for="'+id+'">'+markMatch( data.text, term ) +'</label></div>', { id: id });
+						return markup;
+					}
+
+					var select2_args = $.extend( {
+						templateResult: formatResult,
+						closeOnSelect     : false,
+						allowClear        : $( this ).data( 'allow_clear' ) ? true : false,
+						placeholder       : $( this ).data( 'placeholder' ),
+						minimumInputLength: $( this ).data( 'minimum_input_length' ) ? $( this ).data( 'minimum_input_length' ) : '3',
+						escapeMarkup      : function( m ) {
+							return m;
+						},
+						width: '100%',
+						multiple: true,
+						ajax: {
+							url:         wc_enhanced_select_params.ajax_url,
+							dataType:    'json',
+							delay:       250,
+							data:        function( params ) {
+								query = params;
+								return {
+									term:     params.term,
+									hide_empty: false,
+									action:   'woocommerce_json_search_categories',
+									security: wc_enhanced_select_params.search_categories_nonce
+								};
+							},
+							processResults: function( data ) {
+								var terms = [];
+								if ( data ) {
+									$.each( data, function( id, term ) {
+										terms.push({
+											id:   'id' === return_format ? term.term_id : term.slug,
+											text: term.formatted_name
+										});
+									});
+								}
+								return {
+									results: terms
+								};
+							},
+							cache: true
+						}
+					}, getEnhancedSelectFormatString() );
+
+					$( this ).selectWoo( select2_args ).addClass( 'enhanced' );
```

Comment:
I got totally lost reading this 😳

Can you describe what this does? Maybe increment the comment on line 322 to briefly explain what this is doing.

Reply:
yeah this seemingly was left behind from an old implementation and not necessary anymore.